cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -0,0 +1,879 @@
+            }
+            _ => unimplemented!("baldrdash callee saved on non-i64 reg classes"),
+        };
+    }
+
+    match r.get_class() {
+        RegClass::I64 => {
+            // x0 - x17 inclusive are caller-saves.
+            r.get_hw_encoding() <= 17
+        }
+        RegClass::V128 => {
+            // v0 - v7 inclusive and v16 - v31 inclusive are caller-saves.
+            r.get_hw_encoding() <= 7 || (r.get_hw_encoding() >= 16 && r.get_hw_encoding() <= 31)
+        }
+        _ => panic!("Unexpected RegClass"),
+    }
+}
+
+fn get_caller_saves_set(call_conv: isa::CallConv) -> Set<Writable<Reg>> {
+    let mut set = Set::empty();
+    for i in 0..29 {
+        let x = writable_xreg(i);
+        if is_caller_save(call_conv, x.to_reg().to_real_reg()) {
+            set.insert(x);
+        }
+    }
+    for i in 0..32 {
+        let v = writable_vreg(i);
+        if is_caller_save(call_conv, v.to_reg().to_real_reg()) {
+            set.insert(v);
+        }
+    }
+    set
+}
+
+impl ABIBody for AArch64ABIBody {
+    type I = Inst;
+
+    fn liveins(&self) -> Set<RealReg> {
+        let mut set: Set<RealReg> = Set::empty();
+        for arg in &self.sig.args {
+            if let &ABIArg::Reg(r, _) = arg {
+                set.insert(r);
+            }
+        }
+        set
+    }
+
+    fn liveouts(&self) -> Set<RealReg> {
+        let mut set: Set<RealReg> = Set::empty();
+        for ret in &self.sig.rets {
+            if let &ABIArg::Reg(r, _) = ret {
+                set.insert(r);
+            }
+        }
+        set
+    }
+
+    fn num_args(&self) -> usize {
+        self.sig.args.len()
+    }
+
+    fn num_retvals(&self) -> usize {
+        self.sig.rets.len()
+    }
+
+    fn num_stackslots(&self) -> usize {
+        self.stackslots.len()
+    }
+
+    fn gen_copy_arg_to_reg(&self, idx: usize, into_reg: Writable<Reg>) -> Inst {
+        match &self.sig.args[idx] {
+            &ABIArg::Reg(r, ty) => Inst::gen_move(into_reg, r.to_reg(), ty),
+            &ABIArg::Stack(off, ty) => load_stack(off + 16, into_reg, ty),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn gen_copy_reg_to_retval(&self, idx: usize, from_reg: Reg) -> Inst {
+        match &self.sig.rets[idx] {
+            &ABIArg::Reg(r, ty) => Inst::gen_move(Writable::from_reg(r.to_reg()), from_reg, ty),
+            &ABIArg::Stack(off, ty) => store_stack(off + 16, from_reg, ty),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn gen_ret(&self) -> Inst {
+        Inst::Ret {}
+    }
+
+    fn gen_epilogue_placeholder(&self) -> Inst {
+        Inst::EpiloguePlaceholder {}
+    }
+
+    fn set_num_spillslots(&mut self, slots: usize) {
+        self.spillslots = Some(slots);
+    }
+
+    fn set_clobbered(&mut self, clobbered: Set<Writable<RealReg>>) {
+        self.clobbered = clobbered;
+    }
+
+    fn load_stackslot(
+        &self,
+        slot: StackSlot,
+        offset: usize,
```

Comment:
Please take `u64` or `i64` instead. This is platform dependent, so won't work nice for cross-compilation.

Reply:
Will use `u32`, actually; we don't expect a stackframe to be larger than 4GB.